### PR TITLE
feat: Complete course table view, animation and connect to data layer

### PR DESCRIPTION
**Pull request by Dao-you**
Define course table block component, illustrate background and arrange courses to the table, fulfill and replace placeholder on current course table screen.

~~Brfore repository layer complete, use the temp datatype below:~~ (Completed in #170)
```dart
/// Temporary UI contract for one course entry in the course table.
///
/// Field names and types align with current database schema as much as possible
/// so repository implementation can migrate without API changes.
typedef CourseTableInfoObject = ({
  /// [CourseOfferings.number].
  String number,

  /// [Courses.nameZh].
  String? courseNameZh,

  /// [Teachers.nameZh] of this offering.
  ///
  /// A course offering can have multiple teachers.
  List<String> teacherNamesZh,

  /// [Courses.credits].
  double credits,

  /// [Courses.hours].
  int hours,

  /// [Classrooms.nameZh] of this offering.
  ///
  /// A course offering can use multiple classrooms.
  List<String> classroomNamesZh,

  /// Raw schedule format from [Schedules] table.
  ///
  /// Each entry is one `(dayOfWeek, period)` slot.
  List<({DayOfWeek dayOfWeek, Period period})> schedule,

  /// [Classes.nameZh] of this offering.
  ///
  /// A course offering can target multiple classes.
  List<String> classNamesZh,
});

/// Temporary UI contract for one renderable time block in the course table.
typedef CourseTableBlockObject = ({
  /// Course metadata shown in this block.
  CourseTableInfoObject courseInfo,

  /// Weekday of this block.
  DayOfWeek dayOfWeek,

  /// Inclusive start slot of this block.
  Period startSection,

  /// Inclusive end slot of this block.
  Period endSection,
});


typedef CourseTableSummaryObject = ({
  // The semester this course table belongs to.
  Semester semester,

  /// Course blocks to render in the table.
  List<CourseTableBlockObject> courses,

  /// Whether the table has courses in the morning (before 12:00).
  bool hasAmCourse,
  bool hasNCourse,
  bool hasPmCourse,
  bool hasNightCourse,
  Period earliestStartSection,
  Period latestEndSection,
  bool hasSatCourse,
  bool hasSunCourse,

  /// Total credits of all courses in the table.
  double totalCredits,

  /// Total hours of all courses in the table.
  int totalHours,
});
```

<table>
    <tr>
      <td>
        Screenshot_20260319_192546
      </td>
      <td>
        Screenshot_20260319_192552
      </td>
    </tr>
  </table>

## TODOs

- [x] design course table block component
- [x] define datatype and start session to implement repo layer, thanks to @rileychh ! in #168 ,  #170 
- [x] illustrate course table
- [x] a placeholder bottomsheet on cell click
- [x] match the final datatype from repo layer
- [x] use real data
- [x] sleep first

## Test Plan

- [x] check all UI elements are correct
- [x] works on physical device on Android 16 OneUI 8.0
- [x] works on AVD Android 9.0 Pixel 3 XL
- [x] works on physical iOS 26.3 iPhone 11
- [x] test UI with mock data in different edge-case scenarios:
	* [x] classes only in the afternoon and evening
	* [x] classes only in the morning
	* [x] classes only on weekends (Saturday and Sunday)
	* [x] classes only on Saturdays and weekdays
	* [x] classes on weekdays and weekends (Saturday and Sunday)
	* [x] have classes during N periods

## Follow-up After

- daily view instead of weekly
- replace placefolder info in course detail bottomsheet, with swipe-up for more(outline, classmate, cource materials etc.)
- discuss about action button UI
- no-time course and table matadata display
- adjust font size in table cell, especially in english ✅ #216
- add pull-to-refresh feature in course table ✅ #215

### CODE REVIEW

**Review by rileychh — COMMENTED** [review]
在 Review 的過程中，我自己做了一些修改：

- 將星期標籤從 enum 硬編碼改為 slang i18n 字串，支援中英文切換
- 將課表 provider 移至 `course_table/` 目錄，並修正未 await 的非同步呼叫
- 將 `CourseTableGrid` 的 viewport 參數改為 required non-nullable
- 還原 enhanced enum 的手動格式化設定

File: lib/screens/main/course_table/course_table_cell.dart [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+          width: 50,
+          height: 136,
+          child: CourseTableCellSkeleton(),
+        ),
+      ],
+    ),
+  );
+}
+
+final CourseTableCellData _previewNamedCourseTableCell = (
```
[the remark this answers is not in the document]
Comment (Dao-you) [reply lib/screens/main/course_table/course_table_cell.dart]: mock data 應該從 servise 層實作，回傳的是一整個用戶端的資料，大概不會回傳某一個課程的資料物件。我們可以從假資料中撈前幾筆資料出來 Preview，但這無法很好地應對開發過程中需要修改以檢視各種不同情況的需求。因此我不會將假資料實作套用在這裡。
不過將變數改成 const 是個好建議。

File: lib/screens/main/course_table/course_table_detail_sheet.dart [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+
+    return SafeArea(
+      top: false,
+      child: Padding(
+        padding: .fromLTRB(16, 8, 16, 16),
+        child: SizedBox(
+          width: double.infinity,
```
[the remark this answers is not in the document]
Comment (Dao-you) [reply lib/screens/main/course_table/course_table_detail_sheet.dart]: 已刪

File: lib/screens/main/course_table/course_table_grid.dart [hunk 3]
```diff
@@ -0,0 +1,576 @@
...
+import 'package:flutter/widget_previews.dart';
+import 'package:auto_size_text/auto_size_text.dart';
+import 'package:tattoo/components/widget_preview_frame.dart';
+import 'package:tattoo/i18n/strings.g.dart';
+import 'package:tattoo/models/course.dart';
+import 'package:tattoo/repositories/course_repository.dart';
+import 'package:tattoo/screens/main/course_table/course_table_cell.dart';
+import 'package:tattoo/screens/main/course_table/course_table_detail_sheet.dart';
+
+typedef GridRange = ({
```
[the remark this answers is not in the document]
Comment (Dao-you) [reply lib/screens/main/course_table/course_table_grid.dart]: 4c1255bce0b718fc537cd3f6ea4395a6874776ad

**Review by rileychh — APPROVED** [review]
